### ISSUE TITLE
"az.plot_trace" doesn't work

### ISSUE THREAD

**Issue by liseeocean** [arviz-devs/arviz#1095]
I run the pyro example and find "az.plot_trace(pyro_data) " doesn't work:

UserWarning: Argument backend_kwargs has not effect in matplotlib.plot_distSupplied value won't be used
  "Argument backend_kwargs has not effect in matplotlib.plot_dist"

**Comment by OriolAbril** [comment]
Hi! Thanks for reporting. Could you share the error message (the message you shared only has warnings) and the ArviZ version used? We currently don't have enough info to suggest a workaround nor attempt a fix.

The warnings should not be there though. They have been fixed but the fix has not been released yet, using the [development version](https://github.com/arviz-devs/arviz/blob/master/README.md#development) should remove the warnings.

**Comment by liseeocean** [comment]
Thank you for your quick reply. I try it again and add the "plt.show()". This time the figure can be showed.  Just like as follows:
pyro_data = az.from_pyro(mcmc, prior=prior, posterior_predictive=posterior_predictive,
                         coords={'school': np.arange(eight_school_data['J'])},
                         dims={'theta': ['school']})
pyro_data
az.plot_trace(pyro_data)
plt.show()

**Comment by liseeocean** [comment]
How can I compute log-likelihood in pyro using arviz ?

**Comment by OriolAbril** [comment]
I think `from_pyro` automatically stores log likelihood values for posterios samples _if possible_, in some cases there are issues with vectorization and it is not possible though.

(assuming you are using ArviZ development version) If you do not have a log_likelihood group in your `pyro_data`, thebest place to ask for help is probably [Pyro forums](https://forum.pyro.ai/)

**Comment by liseeocean** [comment]
Hi, I find that the pyrodata will automatically stores log-likelihood using "8 school" example in arviz(pyro) but using my code doesn't have. My code is as follows:
`import matplotlib.pyplot as plt
import numpy as np
#import matplotlib as mpl
import pyro.distributions as pdist
import pyro
import torch
# Initialize random number generator

np.random.seed(123)
plt.style.use('ggplot')

pyro.enable_validation(True)

pyro.set_rng_seed(0)

# True parameter values
alpha, sigma = 1, 1
beta = [1, 2.5]

# Size of dataset
size = 100


# Predictor variable
X1 = np.random.randn(size)
X2 = np.random.randn(size) * 0.2

# Simulate outcome variable
Y = alpha + beta[0]*X1 + beta[1]*X2 + np.random.randn(size)*sigma

#Plot the predictors and Y
fig, axes = plt.subplots(1, 2, sharex=True, figsize=(10,4))
axes[0].scatter(X1, Y)
axes[1].scatter(X2, Y)
axes[0].set_ylabel('Y'); axes[0].set_xlabel('X1'); axes[1].set_xlabel('X2')


def pyro_model(x1, x2, y=None):
    alpha = pyro.sample('alpha', pdist.Normal(0, 10))
    beta = pyro.sample('beta', pdist.Normal(torch.zeros(2,), torch.ones(2,) * 10.))
    sigma = pyro.sample('sigma', pdist.HalfNormal(1.))

    # Expected value of outcome
    mu = alpha + beta[0] * x1 + beta[1] * x2

    # Likelihood (sampling distribution) of observations
    return pyro.sample('y_obs', pdist.Normal(mu, sigma), obs=y)

import arviz as az

from pyro.infer import MCMC, NUTS, Predictive

model=pyro_model

X1=torch.FloatTensor(X1)
X2=torch.FloatTensor(X2)
Y=torch.FloatTensor(Y)

nuts_kernel = NUTS(model, jit_compile=False, ignore_jit_warnings=True)
mcmc = MCMC(nuts_kernel, num_samples=500, warmup_steps=100,
            num_chains=1)

mcmc.run(X1,X2,Y)

posterior_samples = mcmc.get_samples()

posterior_predictive = Predictive(model, posterior_samples).get_samples(
    X1, X2)
prior = Predictive(model, num_samples=500).get_samples(
    X1, X2)

pyro_data = az.from_pyro(
    mcmc,
    prior=prior,
    posterior_predictive=posterior_predictive)

pyro_data
print(pyro_data)`

**Comment by OriolAbril** [comment]
It looks like there are some issues when trying to retrieve the trace. `from_pyro` tries to access the values, but in many cases it is not possible, in which case the try except catches the error and omits log_likelihood group. Would it be more clear if a warning was shown when this happens?

I am not a pyro expert, it can probably be optimized further, the code below seems to do the trick and allows `from_pyro` to retrieve log likelihood data (posting only relevant part):

```
def pyro_model(n, x1, x2, y=None):
    alpha = pyro.sample('alpha', pdist.Normal(0, 10))
    beta1 = pyro.sample('beta1', pdist.Normal(0, 10.))
    beta2 = pyro.sample('beta2', pdist.Normal(0, 10.))
    sigma = pyro.sample('sigma', pdist.HalfNormal(1.))

    with pyro.plate("n", n):
        # Expected value of outcome
        mu = alpha + beta1 * x1 + beta2 * x2

        # Likelihood (sampling distribution) of observations
        return pyro.sample('y_obs', pdist.Normal(mu, sigma), obs=y)

import arviz as az

from pyro.infer import MCMC, NUTS, Predictive

model = pyro_model

X1 = torch.FloatTensor(X1)
X2 = torch.FloatTensor(X2)
Y = torch.FloatTensor(Y)

nuts_kernel = NUTS(model, jit_compile=False, ignore_jit_warnings=True)
mcmc = MCMC(nuts_kernel, num_samples=500, warmup_steps=100,
            num_chains=1)

mcmc.run(size, X1, X2, Y)

posterior_samples = mcmc.get_samples()

posterior_predictive = Predictive(model, posterior_samples).get_samples(
    size, X1, X2)
prior = Predictive(model, num_samples=500).get_samples(
    size, X1, X2)

pyro_data = az.from_pyro(
    mcmc,
    prior=prior,
    posterior_predictive=posterior_predictive
)
pyro_data
print(pyro_data)
print("ok")
```

If you were to need a more detailed explanation on why is this happening or further guidance on how to improve and optimize this, [pyro forums](https://forum.pyro.ai/) is the place to go, I'm afraid my pyro understanding isn't much deeper than that. Hope this helps!

**Comment by liseeocean** [comment]
Thank you very much. Actually, don't have any warning running previous codes.  I have running your improved codes and it works. 
Thank you again!

**Comment by OriolAbril** [comment]
There is no warning currently, but we could add it if it would have reduced your confusion as to why there was no likelihood data, hence the question. Glad to hear it works fine now

**Comment by liseeocean** [comment]
Hi, I find the log_likelihood that stores in pyro_data is a big array [num_iteraton,num_datalength] with your improved codes. I think it return logp(xi|theta) for a log_likelihood for each observed xi not L=log[p(x1|theta)*p(x2|theta)...p(xn|theta)]. I am not sure I have understood it rightly but it is a problem.

**Comment by OriolAbril** [comment]
AtviZ stores the pointwise log likelihood, to get the log likelihood for all observations you have to sum over the observation axis.
